internal/util/get/get.go

```diff
@@ -257,22 +256,15 @@ func (c *Command) DefaultValues() error {
 // cloneFrom values.
 func (c *Command) upsertKptfile(spec *git.RepoSpec) error {
 	// read KptFile cloned with the package if it exists
-	kpgfile, err := kptfileutil.ReadFile(c.Destination)
+	var contents []byte
+	contents, err := ioutil.ReadFile(filepath.Join(c.Destination, kptfile.KptFileName))
 	if err != nil {
-		// no KptFile present, create a default
-		kpgfile = kptfile.KptFile{
-			ResourceMeta: yaml.ResourceMeta{
-				TypeMeta: yaml.TypeMeta{
-					APIVersion: kptfile.TypeMeta.APIVersion,
-					Kind:       kptfile.TypeMeta.Kind,
-				},
-				ObjectMeta: yaml.ObjectMeta{
-					NameMeta: yaml.NameMeta{
-						Name: c.Name,
-					},
-				},
-			},
-		}
+		contents = []byte(fmt.Sprintf(kptfileTemplate, c.Name))
+	}
+
+	kfNode, err := yaml.Parse(string(contents))
```

Comment:
I think requiring every part of kpt that needs to manipulate the Kptfile to fall back to kyaml is too awkward. I think we should still preserve the `pkgfile.Kptfile` type, but just use it as a wrapper around the `RNode` data structure. And we can provide functions that allow other parts of kpt to still pass in and fetch data structure like `pkgfile.Upstream`. To the extent possible, the fact that we are not unmarshalling the yaml into structs should be an implementation detail.